### ISSUE TITLE
[ENHANCEMENT] Add bonding and atom information to MDANSE

### ISSUE THREAD

**Issue by ChiCheng45** [ISISNeutronMuon/MDANSE#286]
**Is your feature request related to a problem? Please describe.**
At the moment there is no bonding and atom information in MDANSE. It might be useful to be able to distinguish between different bonding types (single, double, triple, ...) and atoms (aromatic, aliphatic, ...)in the future. This might be important if we want to allow the user to make more complex atom selections.

**Describe the solution you'd like**
The way bonding in MDANSE needs to be refactored and a bonding type needs to be added to the chemical entities.

**Comment by ChiCheng45** [comment]
Bonding types could be determined with rdkit see #350.

**Comment by oerc0122** [comment]
Is this done now?

**Comment by ChiCheng45** [comment]
> Is this done now?

No, this isn't finished yet because all bond types in MDANSE are unspecified. The SMARTS pattern matching doesn't work with some patterns that specify bond and atom types, etc.
